extractor/src/main/java/org/schabi/newpipe/extractor/services/soundcloud/extractors/SoundcloudStreamExtractor.java

```diff
@@ -190,54 +193,102 @@ public String getHlsUrl() {
+                    // TODO: move this to a separate method to generate valid urls when needed (e.g. resuming a paused stream)
+
+                    if (t.getObject("format").getString("protocol").equals("progressive")) {
                         // This url points to the endpoint which generates a unique and short living url to the stream.
-                        // TODO: move this to a separate method to generate valid urls when needed (e.g. resuming a paused stream)
                         url += "?client_id=" + SoundcloudParsingHelper.clientId();
                         final String res = dl.get(url).responseBody();
 
                         try {
                             JsonObject mp3UrlObject = JsonParser.object().from(res);
                             // Links in this file are also only valid for a short period.
-                            audioStreams.add(new AudioStream(mp3UrlObject.getString("url"),
-                                    MediaFormat.MP3, 128));
-                        } catch (JsonParserException e) {
+                            mediaUrl = mp3UrlObject.getString("url");
+                        } catch (final JsonParserException e) {
                             throw new ParsingException("Could not parse streamable url", e);
                         }
+                    } else if (t.getObject("format").getString("protocol").equals("hls")) {
+                        // This url points to the endpoint which generates a unique and short living url to the stream.
+                        url += "?client_id=" + SoundcloudParsingHelper.clientId();
+                        final String res = dl.get(url).responseBody();
+
+                        try {
+                            final JsonObject mp3HlsUrlObject = JsonParser.object().from(res);
+                            // Links in this file are also only valid for a short period.
+                            try {
+                                mediaUrl = getUrlFromHlsManifest(mp3HlsUrlObject.getString("url"));
+                            } catch (final ParsingException e) {
+                                continue;
+                            }
+                        } catch (final JsonParserException e) {
+                            throw new ParsingException("Could not parse streamable url", e);
+                        }
+                    } else {
+                        continue;
                     }
+
+                    audioStreams.add(new AudioStream(mediaUrl, mediaFormat, bitrate));
                 }
             }
 
-        } catch (NullPointerException e) {
+        } catch (final NullPointerException e) {
             throw new ExtractionException("Could not get SoundCloud's track audio url", e);
         }
 
-        if (audioStreams.isEmpty()) {
-            throw new ContentNotSupportedException("HLS audio streams are not yet supported");
-        }
-
```

Comment:
For debugging purposes I suggest you to revert this, so that you are instantly notified if `audioStreams.isEmpty()` is `true`